src/Orc.Theming.Tests/ExtensionMethods/ControlExtensionsTests.cs

```diff
@@ -0,0 +1,66 @@
+﻿namespace Orc.Theming.Tests.Services;
+
+using System;
+using System.Threading;
+using System.Windows;
+using System.Windows.Controls;
+using NUnit.Framework;
+
+[TestFixture]
+public class ControlExtensionsTests
+{
+    [Test]
+    [Apartment(ApartmentState.STA)]
+    public void Get_Required_Template_Should_Return_Template_If_It_Exist()
+    {
+        const string partName = "PART_Border";
+
+        //Prepare
+        var button = PrepareControlWithBorderPart(partName);
+
+        //Act
+        var border = button.GetRequiredTemplateChild<Border>(partName);
+
+        Assert.That(border, Is.Not.Null);
+    }
+
+    [Test]
+    [Apartment(ApartmentState.STA)]
+    public void Get_Required_Template_Should_Fail_With_InvalidOperationException_If_Template_Does_Not_Exist()
+    {
+        const string partName = "PART_Border";
+        const string differentPartName = partName + "1";
+
+        //Prepare
+        var button = PrepareControlWithBorderPart(partName);
+
+        //Act - Assert
+        Assert.Throws<InvalidOperationException>(() => button.GetRequiredTemplateChild<Border>(differentPartName));
```

Comment:
First of all, compliments for writing these tests!

I recently started validating the *message* of the exception as well during unit tests. E.g. something similar to:

```
var exception = Assert.Throws<InvalidOperationException>(() => button.GetRequiredTemplateChild<Border>(differentPartName));

Assert.That(exception!.Message, Is.Equal.To("Can't find template part ..."));
```